### ISSUE TITLE
V2: infinite loop on recursive type when referenced by media type

### ISSUE THREAD

**Issue by Smerom** [goadesign/goa#1859]
Maybe related to #613

## Expected Behavior
Should be able to create media types from recursive structures.

## Actual Behavior
Get the following stack overflow:

```
exit status 2
runtime: goroutine stack exceeds 1000000000-byte limit
fatal error: stack overflow

runtime stack:
runtime.throw(0x1549dd8, 0xe)
	/usr/local/Cellar/go/1.9.2/libexec/src/runtime/panic.go:605 +0x95
runtime.newstack(0x0)
	/usr/local/Cellar/go/1.9.2/libexec/src/runtime/stack.go:1050 +0x6e1
runtime.morestack()
	/usr/local/Cellar/go/1.9.2/libexec/src/runtime/asm_amd64.s:415 +0x86

goroutine 1 [running]:
goa.design/goa/design.(*AttributeExpr).shouldInherit(0xc42028a780, 0xc42028a500, 0x0)
	/Users/qm/go/src/goa.design/goa/design/attribute.go:535 +0x9a fp=0xc4402bc350 sp=0xc4402bc348 pc=0x1182eca
goa.design/goa/design.(*AttributeExpr).inheritRecursive(0xc42028a780, 0xc42028a500)
	/Users/qm/go/src/goa.design/goa/design/attribute.go:501 +0x3f fp=0xc4402bc3d0 sp=0xc4402bc350 pc=0x1182a7f
goa.design/goa/design.(*AttributeExpr).Inherit(0xc42028a780, 0xc42028a500)
	/Users/qm/go/src/goa.design/goa/design/attribute.go:282 +0xcc fp=0xc4402bc408 sp=0xc4402bc3d0 pc=0x11814fc
goa.design/goa/design.(*AttributeExpr).inheritRecursive(0xc42028a780, 0xc42028a500)
	/Users/qm/go/src/goa.design/goa/design/attribute.go:518 +0x25f fp=0xc4402bc488 sp=0xc4402bc408 pc=0x1182c9f
goa.design/goa/design.(*AttributeExpr).Inherit(0xc42028a780, 0xc42028a500)
	/Users/qm/go/src/goa.design/goa/design/attribute.go:282 +0xcc fp=0xc4402bc4c0 sp=0xc4402bc488 pc=0x11814fc
goa.design/goa/design.(*AttributeExpr).inheritRecursive(0xc42028a780, 0xc42028a500)
```
## Steps to Reproduce the Problem (including complete and simple design if relevant)
Minimal design:
```golang
package design

import (
	. "goa.design/goa/http/dsl"
)

var RecursiveExample = Type("RecursiveExample", func() {
	Attribute("rec", "RecursiveExample")
})

var RecMedia = ResultType("RecMedia", func() {
	Reference(RecursiveExample)
	Attribute("rec")
})
```

## Specifications

  - goa Version: V2 at commit f14c160845d974e9cf0a9a01abbca41b2adb9483
  - Platform (Linux/Windows etc.): Mac OSX

**Comment by Smerom** [comment]
It is the `Reference(RecursiveExample)` that appears to be causing the problems.

**Comment by raphael** [comment]
Thank you for the detailed report, this is fixed by #1920
